fsspec/spec.py

```diff
@@ -220,12 +287,22 @@ def open(self, path, mode='rb', block_size=None, **kwargs):
         if 'b' not in mode:
             mode = mode.replace('t', '') + 'b'
             return io.TextIOWrapper(
-                self.open(self, path, mode, block_size, **kwargs))
+                self.open(path, mode, block_size, **kwargs))
+        else:
+            ac = kwargs.pop('autocommit', not self._intrans)
+            f = self._open(path, mode=mode, block_size=block_size,
+                           autocommit=ac, **kwargs)
+            if not ac:
```

Comment:
Should autocommit be necessarily True is `_intrans` isn't set, and should we avoid collecting up instances that might never be cleared?

Reply:
`autocommit` only applies to writing at `__exit__` so i think its a sane default in terms of user expectations

Reply:
Thats more explicit.  👍